cmd/gateclient/client.go

```diff
@@ -96,6 +97,70 @@ func (m *GatewayClient) GateEndpoint() string {
 	return m.Config.Gate.Endpoint
 }
 
+func NewHTTPClient(ctx context.Context, cfg *config.Config) (*http.Client, context.Context, error) {
```

Comment:
The pattern of returning a context from `NewHTTPClient` seems strange to me...generally I'd expect this to accept a context, but not to return one.

Reply:
Yeah, for sure, not ideal. The problem is that the original code uses context state values, which returns context copy.  I was unsure how else to accomplish this without returning the modified context.  Like here:
https://github.com/spinnaker/spin/pull/244/files#diff-146916aad3615e73605759c9ea128128R357

I will think about alternate strategies, not sure we have many other than to capture into some client wrapper struct like the current code does.

Reply:
I separated out the code that modified context state into a single `ContextWithAuth` so now it is more clear without the ctx coping everywhere.